src/main/java/de/tum/in/www1/artemis/domain/FileUploadSubmission.java

```diff
@@ -34,8 +37,25 @@ public String getSubmissionExerciseType() {
     @PostRemove
     public void onDelete() {
         if (filePath != null) {
-            fileService.schedulePathForDeletion(Path.of(filePath), 0);
+            Path actualPath = getActualPathForPublicPath(filePath);
+            fileService.schedulePathForDeletion(actualPath, 0);
+        }
+    }
+
+    private Path getActualPathForPublicPath(String filePath) {
+        // Note: This method duplicates functionality from actualPathForPublicPath on FilePathService
```

Comment:
Would it be possible to make the relevant method in `FilePathService` `static`? I think then a dependency on the service only by calling the static method should be fine here.

Reply:
There is really no reason to not make it static since FilePathService doesn't depend on any other services. If a static method is fine I'll try it out.